### PR TITLE
Use STM32F303xE for PDM

### PR DESCRIPTION
**Pull request by akrantz01**
### Summary
Switches from the STM32F302xC to the STM32F303xE for the PDM.

### Changelist 
- Updated and re-generated the CubeMX project
- Moved the shared SPI code to the BMS

### Testing Done
N/A

### Resolved Issues
N/A

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/PDM/Inc/stm32f3xx_hal_conf.h [hunk 1]
```diff
@@ -71,7 +71,7 @@
 /*#define HAL_LPTIM_MODULE_ENABLED   */
 /*#define HAL_RNG_MODULE_ENABLED   */
 /*#define HAL_RTC_MODULE_ENABLED   */
-#define HAL_SPI_MODULE_ENABLED
+/*#define HAL_SPI_MODULE_ENABLED   */
```
Comment (pasitlaothamatas) [inline boards/PDM/Inc/stm32f3xx_hal_conf.h:74]: if we remove this from cube and delete the efuse code with shared SPI, can we keep shared spi in the shared folder?

**Comment by pasitlaothamatas** [comment]
@joethrstn if you have time do you wanna go over the cube config to check if the pins are configured properly?

File: boards/PDM/Inc/main.h [hunk 2]
```diff
@@ -88,78 +82,60 @@ extern "C"
...
-#define CUR_SYNC_DI_FL_DI_FR_Pin GPIO_PIN_10
-#define CUR_SYNC_DI_FL_DI_FR_GPIO_Port GPIOA
-#define CSB_DI_FL_DI_FR_Pin GPIO_PIN_15
-#define CSB_DI_FL_DI_FR_GPIO_Port GPIOA
-#define FSB_DI_FL_DI_FR_Pin GPIO_PIN_10
-#define FSB_DI_FL_DI_FR_GPIO_Port GPIOC
-#define FSOB_DI_FL_DI_FR_Pin GPIO_PIN_11
-#define FSOB_DI_FL_DI_FR_GPIO_Port GPIOC
-#define PGOOD_Pin GPIO_PIN_12
+#define FR_STBY_DI_REAR_Pin GPIO_PIN_1
```
Comment (pasitlaothamatas) [inline boards/PDM/Inc/main.h:85]: @akrantz01 is this updated for the new PDM or is this just ported from the old PDM? schematic for the new PDM pin outs could be a little different iirc

Comment (akrantz01) [reply boards/PDM/Inc/main.h:85]: I'm not super sure. I've been going off a PDF @markomarkusi gave me a few weeks ago

Comment (pasitlaothamatas) [reply boards/PDM/Inc/main.h:85]: can you message joe your email and he'll try to get you set up with altium

**Comment by timbitss** [comment]
inactive

**Pull request closed without merging**